workflow/libero_flow_poc/libero_flow/decider.py

```diff
@@ -0,0 +1,178 @@
+import json
+from time import gmtime, strftime
+from typing import Any, Dict
+import uuid
+
+import pika
+
+from libero_flow.state_utils import (
+    CANCELLED,
+    FINISHED,
+    IN_PROGRESS,
+    PENDING,
+    PERMANENT_FAILURE,
+    SCHEDULED,
+    SUCCEEDED,
+    TEMPORARY_FAILURE,
+)
+from libero_flow.event_utils import (
+    get_channel,
+    message_handler,
+    setup_exchanges_and_queues,
+    DELIVERY_MODE_PERSISTENT,
+    DECISION_RESULT_EXCHANGE,
+    DECISION_RESULT_QUEUE,
+    SCHEDULED_DECISION_QUEUE,
+)
+from libero_flow.state_utils import get_workflow_state
+
+
+def send_decision_message(decision: Dict):
+    """create and send decision task message.
+
+    :param decision: dict
+    :return:
+    """
+    with get_channel() as channel:
+        message = {
```

Comment:
same as previous comment - no need to force messages into the bus schema if these messages are internal to this component.

It may be useful for the bus to separately receive events about what happens inside the worfklow of course.

Reply:
👍